src/Stratis.Bitcoin.Features.Wallet/WalletRPCController.cs

```diff
@@ -197,6 +202,77 @@ public decimal GetBalance(string accountName, int minConfirmations = 0)
             return balance?.ToUnit(MoneyUnit.BTC) ?? 0;
         }
 
+        [ActionName("listsinceblock")]
+        [ActionDescription("Get all transactions in blocks since block 'blockhash', or all transactions if omitted.")]
+        public async Task<ListSinceBlockModel> ListSinceBlockAsync(string blockHash, int targetConfirmations = 1)
+        {
+            WalletAccountReference accountReference = this.GetWalletAccountReference();
+            Wallet wallet = this.walletManager.GetWallet(accountReference.WalletName);
+            Block block = null;
+
+            if (!string.IsNullOrEmpty(blockHash) && uint256.TryParse(blockHash, out uint256 hashBlock))
+            {
+                this.ConsensusManager.GetOrDownloadBlocks(new List<uint256> { hashBlock }, b => { block = b.Block; });
+            }
+
+            IEnumerable<TransactionData> transactions = wallet.GetAllTransactions();
+
+            var model = new ListSinceBlockModel();
+
+            foreach (TransactionData transactionData in transactions)
+            {
+                GetTransactionModel transaction = await this.GetTransactionAsync(transactionData.Id.ToString());
+
+                int blockHeight = transactionData.BlockHeight ?? 0;
+
+                if (block != null && blockHeight < this.ChainIndexer.GetHeader(block?.GetHash()).Height)
```

Comment:
This is waistfull, you can instead vat the method above GetBlock also fetch the header